### DISCUSSION TITLE
External Dependencies

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [SamErde/PSPreworkout#5] [General]
The `Edit-WingetSettings` function is calling `Get-WinGetSettings` which I'm assuming is the command in the `Microsoft.Winget.Client` module. If this is a dependency, you should add it to the module manifest.

```powershell
ExternalModuleDependencies = @('Microsoft.Winget.Client')
```

This will force the download of this module when the user installs your module.

**Comment by SamErde** [comment]
Noted. Thanks! Wonder if I could simply reference the expected path to avoid creating hard dependencies. Then again, this function is only useful if WinGet is installed, so... 🤔

**Comment by jdhitsolutions** [comment]
You can add code in `Edit-WingetSettings` to test if `winget` is installed. Or test for the settings file. You don't even need `Get-WinGetSettings`

```powershell
 Test-path "$env:LOCALAPPDATA\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json"
```

**Reply by SamErde** [reply]
I opted for this approach instead of declaring a dependency.

**Reply by jdhitsolutions** [reply]
Good call. I try to avoid dependencies in my modules, especially when there is an easy alternative.

**Comment by SamErde** [comment]
After our other conversations, I feel like it might be more precise to rename this to `Edit-WinGetSettingsFile`.

**Comment by SamErde** [comment]
TIL about the `winget settings` command, which opens WinGet's settings.json file in your default JSON editor. 😆
